beacon_node/http_api/src/attester_duties.rs

```diff
@@ -208,27 +215,58 @@ fn convert_to_api_response<T: BeaconChainTypes>(
         )));
     }
 
+    // need effective balances for epoch before the requested epoch
+    let relevant_balances_epoch = request_epoch.saturating_sub(1u64);
+    chain
+        .effective_balances_cache
+        // have to load the effective balances cache for the previous epoch
+        .load_epoch(chain, relevant_balances_epoch)
+        .map_err(warp_utils::reject::beacon_chain_error)?;
+
     let usize_indices = indices.iter().map(|i| *i as usize).collect::<Vec<_>>();
     let index_to_pubkey_map = chain
         .validator_pubkey_bytes_many(&usize_indices)
         .map_err(warp_utils::reject::beacon_chain_error)?;
 
-    let data = duties
-        .into_iter()
-        .zip(indices)
-        .filter_map(|(duty_opt, &validator_index)| {
-            let duty = duty_opt?;
-            Some(api_types::AttesterData {
-                pubkey: *index_to_pubkey_map.get(&(validator_index as usize))?,
-                validator_index,
-                committees_at_slot: duty.committees_at_slot,
-                committee_index: duty.index,
-                committee_length: duty.committee_len as u64,
-                validator_committee_index: duty.committee_position as u64,
-                slot: duty.slot,
-            })
-        })
-        .collect::<Vec<_>>();
+    let mut data = Vec::new();
+    for (duty_opt, &validator_index) in duties.into_iter().zip(indices.iter()) {
+        let (duty, pubkey) = match (
+            duty_opt,
+            index_to_pubkey_map.get(&(validator_index as usize)),
+        ) {
+            (Some(duty), Some(pubkey)) => (duty, pubkey),
+            _ => continue,
+        };
+
+        let committee_length =
+            if chain.spec.fork_name_at_slot::<T::EthSpec>(duty.slot) >= ForkName::Deneb {
+                let beacon_committee = state
+                    .get_beacon_committee(duty.slot, duty.index)
+                    .map_err(warp_utils::reject::beacon_state_error)?;
+                let total_committee_balance = chain
+                    .effective_balances_cache
+                    .get_committee_balance::<T::EthSpec>(&beacon_committee)
+                    .map_err(warp_utils::reject::beacon_chain_error)?;
```

Comment:
This looks like it will frequently error with `Error::CacheMiss`

Reply:
Why? The epoch cache is loaded just above?